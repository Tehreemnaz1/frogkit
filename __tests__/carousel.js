import { mount, shallow } from 'vue-test-utils'
import Vue from 'vue'
import Carousel from '@/components/carousel'

const propsData = {
  slides: Array.from({length: 3}).map((_,i) => `https://unsplash.it/320?random&${i}`),
  loop: true,
  active: 1
}

describe('Carousel component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = mount(Carousel, { propsData })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should move to the right slide without looping', () => {
    const cmp = shallow(Carousel, { propsData: { ...propsData, loop: false } })
    expect(cmp.vm.selected).toBe(1)
    cmp.vm.moveTo(2)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(propsData.slides.length)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(-1)
    expect(cmp.vm.selected).toBe(2)
  })

  it('Should move to the right slide when looping', () => {
    const cmp = shallow(Carousel, { propsData })
    expect(cmp.vm.selected).toBe(propsData.active)
    cmp.vm.moveTo(2)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(propsData.slides.length)
    expect(cmp.vm.selected).toBe(0)
    cmp.vm.moveTo(-1)
    expect(cmp.vm.selected).toBe(propsData.slides.length -1)
  })

  it('Should provide the right styles', () => {
    const cmp = shallow(Carousel, { propsData })
    cmp.vm.moveTo(0)
    expect(cmp.vm.styles).toEqual({ transform: 'translateX(0%)' })
    cmp.vm.moveTo(2)
    expect(cmp.vm.styles).toEqual({ transform: 'translateX(-200%)' })
  })
})
