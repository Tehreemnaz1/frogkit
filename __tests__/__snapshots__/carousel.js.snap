// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component Should render component and match snapshot 1`] = `
<section class="fk-carousel">
    <swipeable>
        <div class="fk-carousel__track"><img src="https://unsplash.it/320?random&amp;0" draggable="false" alt=" 0" class="fk-carousel__slide"><img src="https://unsplash.it/320?random&amp;1" draggable="false" alt=" 1" class="fk-carousel__slide"><img src="https://unsplash.it/320?random&amp;2"
                draggable="false" alt=" 2" class="fk-carousel__slide"></div>
    </swipeable>
    <div class="fk-carousel__indicators"><label class="fk-carousel__indicator"></label><label class="fk-carousel__indicator fk-carousel__indicator--active"></label><label class="fk-carousel__indicator"></label></div>
    <flex justify="space-evenly" class="fk-carousel__previews"><img alt=" 0" src="https://unsplash.it/320?random&amp;0" draggable="false" class="fk-carousel__preview"><img alt=" 1" src="https://unsplash.it/320?random&amp;1" draggable="false" class="fk-carousel__preview fk-carousel__preview--active"><img alt=" 2"
            src="https://unsplash.it/320?random&amp;2" draggable="false" class="fk-carousel__preview"></flex>
</section>
`;
